fix: Handle index value of null in BatchRequest

mihail-ca opened the pull request:
fix: Handle index value of null in BatchRequest

- added exception with detailed message, if there are requests with missing index, in BatchRequest
- added property to control whether to throw an exception, or just log the error and ignore the requests with missing index
- the property added is called bulkRequestIgnoreNullIndex

Closes https://github.com/camunda/camunda/issues/32804

lenaschoenburg committed:
refactor: rewrite `FeelFunctionProvider` in Java

lenaschoenburg committed:
refactor: rewrite `FeelToMessagePackTransformer` in Java

lenaschoenburg committed:
refactor: rewrite `MessagePackContext` in Java

lenaschoenburg committed:
refactor: rewrite `MessagePackValueMapper` in Java

lenaschoenburg committed:
feat: provide a logger for `feel`

lenaschoenburg committed:
build: `feel` integration does not rely on scala anymore

lenaschoenburg committed:
refactor: rewrite `FeelVariableContext` in Java

lenaschoenburg committed:
refactor: rewrite `FeelEvaluationResult` in Java

lenaschoenburg committed:
build: expression language does not rely on scala anymore

lenaschoenburg committed:
build: remove `scala-maven-plugin`

lenaschoenburg committed:
fix: use only argument for an exact error message

kristinkomschow committed:
fix: allowed for empty POST request bodies

made request bodies optional for POST requests to avoid unspecific error message

closes #camunda/operate/2694

kristinkomschow committed:
test: added tests for empty POST request bodies

Commit:
fix: Handle index value of null in BatchRequest

Commit:
fix: Handle index value of null in BatchRequest

Commit:
fix: Handle index value of null in BatchRequest

Commit:
fix: Handle index value of null in BatchRequest

matt-whiteman commented on operate/schema/src/main/java/io/camunda/operate/store/opensearch/client/sync/OpenSearchBatchOperations.java:
```diff
@@ -71,6 +85,64 @@ private void processBulkRequest(OpenSearchClient osClient, BulkRequest bulkReque
     }
   }
 
+  private BulkRequest validateIndices(BulkRequest bulkRequest, boolean ignoreNullIndex)
```
Please consider moving some of these functions into helper classes with unit tests so we avoid having a large monolithic class

matt-whiteman committed:
test: refactor ReindexIT and SchemaCreationIT

matt-whiteman committed:
test: remove some lower-value integration tests from operate

renovate[bot] committed:
deps: Update actions/checkout digest to 1d96c77

renovate[bot] committed:
deps: Update dependency com.amazonaws:aws-java-sdk-bom to v1.12.704

renovate[bot] committed:
deps: Update actions/checkout digest to 1d96c77 (main) (#17589)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| actions/checkout | action | digest | `cd7d8d6` -> `1d96c77` |

---

### Configuration

📅 **Schedule**: Branch creation - "after 10pm every weekday,before 6am
every weekday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/camunda/zeebe).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy4zMDEuNCIsInVwZGF0ZWRJblZlciI6IjM3LjMwMS40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJhdXRvbWVyZ2UiXX0=-->

renovate[bot] committed:
deps: Update dependency com.amazonaws:aws-java-sdk-bom to v1.12.704 (main) (#17590)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.amazonaws:aws-java-sdk-bom](https://aws.amazon.com/sdkforjava)
([source](https://togithub.com/aws/aws-sdk-java)) | `1.12.703` ->
`1.12.704` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/com.amazonaws:aws-java-sdk-bom/1.12.704?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/com.amazonaws:aws-java-sdk-bom/1.12.704?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/com.amazonaws:aws-java-sdk-bom/1.12.703/1.12.704?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/com.amazonaws:aws-java-sdk-bom/1.12.703/1.12.704?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

### Release Notes

<details>
<summary>aws/aws-sdk-java (com.amazonaws:aws-java-sdk-bom)</summary>

###
[`v1.12.704`](https://togithub.com/aws/aws-sdk-java/blob/HEAD/CHANGELOG.md#112704-2024-04-18)

[Compare
Source](https://togithub.com/aws/aws-sdk-java/compare/1.12.703...1.12.704)

#### **Amazon GuardDuty**

-   ### Features
    -   Added IPv6Address fields for local and remote IP addresses

#### **Amazon QuickSight**

-   ### Features
- This release adds support for the Cross Sheet Filter and Control
features, and support for warnings in asset imports for any permitted
errors encountered during execution

#### **Amazon SageMaker Service**

-   ### Features
    -   Removed deprecated enum values and updated API documentation.

#### **Amazon WorkSpaces**

-   ### Features
- Adds new APIs for managing and sharing WorkSpaces BYOL configuration
across accounts.

#### **EMR Serverless**

-   ### Features
- This release adds the capability to publish detailed Spark engine
metrics to Amazon Managed Service for Prometheus (AMP) for enhanced
monitoring for Spark jobs.

#### **Elastic Disaster Recovery Service**

-   ### Features
    -   Outpost ARN added to Source Server and Recovery Instance

#### **IAM Roles Anywhere**

-   ### Features
- This release introduces the PutAttributeMapping and
DeleteAttributeMapping APIs. IAM Roles Anywhere now provides the
capability to define a set of mapping rules, allowing customers to
specify which data is extracted from their X.509 end-entity
certificates.

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "after 10pm every weekday,before 6am
every weekday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/camunda/zeebe).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy4zMDEuNCIsInVwZGF0ZWRJblZlciI6IjM3LjMwMS40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJhdXRvbWVyZ2UiXX0=-->

renovate[bot] committed:
deps: Update dependency org.camunda.feel:feel-engine to v1.17.6

renovate[bot] committed:
deps: Update dependency software.amazon.awssdk:bom to v2.25.34

renovate[bot] committed:
deps: Update dependency org.camunda.feel:feel-engine to v1.17.6 (main) (#17594)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [org.camunda.feel:feel-engine](http://www.camunda.org)
([source](https://togithub.com/camunda/feel-scala)) | `1.17.5` ->
`1.17.6` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.camunda.feel:feel-engine/1.17.6?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.camunda.feel:feel-engine/1.17.6?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.camunda.feel:feel-engine/1.17.5/1.17.6?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.camunda.feel:feel-engine/1.17.5/1.17.6?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

### Release Notes

<details>
<summary>camunda/feel-scala (org.camunda.feel:feel-engine)</summary>

###
[`v1.17.6`](https://togithub.com/camunda/feel-scala/releases/tag/1.17.6)

[Compare
Source](https://togithub.com/camunda/feel-scala/compare/1.17.5...1.17.6)

#### What's Changed :bug:

- \[Backport 1.17] feat: I can interrupt an expression evaluation by
[@&#8203;saig0](https://togithub.com/saig0) in
[https://github.com/camunda/feel-scala/pull/830](https://togithub.com/camunda/feel-scala/pull/830)
- build: Run Java 21 in CI by
[@&#8203;psavidis](https://togithub.com/psavidis) in
[https://github.com/camunda/feel-scala/pull/808](https://togithub.com/camunda/feel-scala/pull/808)

**Full Changelog**:
https://github.com/camunda/feel-scala/compare/1.17.5...1.17.6

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "after 10pm every weekday,before 6am
every weekday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/camunda/zeebe).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy4zMDEuNCIsInVwZGF0ZWRJblZlciI6IjM3LjMwMS40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJhdXRvbWVyZ2UiXX0=-->

renovate[bot] committed:
deps: Update dependency software.amazon.awssdk:bom to v2.25.34 (main) (#17595)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [software.amazon.awssdk:bom](https://aws.amazon.com/sdkforjava) |
`2.25.33` -> `2.25.34` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/software.amazon.awssdk:bom/2.25.34?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/software.amazon.awssdk:bom/2.25.34?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/software.amazon.awssdk:bom/2.25.33/2.25.34?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/software.amazon.awssdk:bom/2.25.33/2.25.34?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

### Configuration

📅 **Schedule**: Branch creation - "after 10pm every weekday,before 6am
every weekday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/camunda/zeebe).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy4zMDEuNCIsInVwZGF0ZWRJblZlciI6IjM3LjMwMS40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJhdXRvbWVyZ2UiXX0=-->

renovate[bot] committed:
deps: Update dependency zod to v3.22.5

renovate[bot] committed:
deps: Update spring boot to v3.2.5

renovate[bot] committed:
deps: Update dependency zod to v3.22.5 (main) (#17596)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [zod](https://zod.dev) ([source](https://togithub.com/colinhacks/zod))
| [`3.22.4` ->
`3.22.5`](https://renovatebot.com/diffs/npm/zod/3.22.4/3.22.5) |
[![age](https://developer.mend.io/api/mc/badges/age/npm/zod/3.22.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/npm/zod/3.22.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/npm/zod/3.22.4/3.22.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/zod/3.22.4/3.22.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

### Release Notes

<details>
<summary>colinhacks/zod (zod)</summary>

###
[`v3.22.5`](https://togithub.com/colinhacks/zod/compare/v3.22.4...e7a9b9b3033991be6b4225f1be21da39c250bbb0)

[Compare
Source](https://togithub.com/colinhacks/zod/compare/v3.22.4...e7a9b9b3033991be6b4225f1be21da39c250bbb0)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "after 10pm every weekday,before 6am
every weekday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/camunda/zeebe).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy4zMDEuNCIsInVwZGF0ZWRJblZlciI6IjM3LjMwMS40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJhdXRvbWVyZ2UiXX0=-->

renovate[bot] committed:
deps: Update spring boot to v3.2.5 (main) (#17597)

[![Mend
Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
|
[org.springframework.boot:spring-boot-maven-plugin](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-maven-plugin/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-maven-plugin/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-maven-plugin/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-maven-plugin/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-starter-webflux](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-starter-webflux/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-starter-webflux/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-starter-webflux/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-starter-webflux/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-starter-test](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-starter-test/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-starter-test/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-starter-test/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-starter-test/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-dependencies](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-dependencies/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-dependencies/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-dependencies/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-dependencies/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-starter-json](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-starter-json/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-starter-json/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-starter-json/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-starter-json/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-starter-actuator](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-starter-actuator/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-starter-actuator/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-starter-actuator/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-starter-actuator/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-configuration-processor](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-configuration-processor/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-configuration-processor/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-configuration-processor/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-configuration-processor/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-starter](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-starter/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-starter/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-starter/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-starter/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-test](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-test/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-test/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-test/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-test/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-actuator-autoconfigure](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-actuator-autoconfigure/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-actuator-autoconfigure/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-actuator-autoconfigure/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-actuator-autoconfigure/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-autoconfigure](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-autoconfigure/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-autoconfigure/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-autoconfigure/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-autoconfigure/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot-actuator](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot-actuator/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot-actuator/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot-actuator/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot-actuator/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[org.springframework.boot:spring-boot](https://spring.io/projects/spring-boot)
([source](https://togithub.com/spring-projects/spring-boot)) | `3.2.4`
-> `3.2.5` |
[![age](https://developer.mend.io/api/mc/badges/age/maven/org.springframework.boot:spring-boot/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/maven/org.springframework.boot:spring-boot/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/maven/org.springframework.boot:spring-boot/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/maven/org.springframework.boot:spring-boot/3.2.4/3.2.5?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

### Release Notes

<details>
<summary>spring-projects/spring-boot
(org.springframework.boot:spring-boot-maven-plugin)</summary>

###
[`v3.2.5`](https://togithub.com/spring-projects/spring-boot/compare/v3.2.4...v3.2.5)

[Compare
Source](https://togithub.com/spring-projects/spring-boot/compare/v3.2.4...v3.2.5)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "after 10pm every weekday,before 6am
every weekday" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about these
updates again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Mend
Renovate](https://www.mend.io/free-developer-tools/renovate/). View
repository job log
[here](https://developer.mend.io/github/camunda/zeebe).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy4zMDEuNCIsInVwZGF0ZWRJblZlciI6IjM3LjMwMS40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJhdXRvbWVyZ2UiXX0=-->

kristinkomschow committed:
fix: allowed for empty POST request bodies (#17450)

## Description
Made request bodies optional for POST requests to avoid unspecific error
messages.

## Related issues

closes
[#camunda/operate/2694](https://github.com/camunda/operate/issues/2694)

ce-dmelnych committed:
feat: extend Execution Listener support to task elements

This commit integrates ELs support across all task elements within the Zeebe workflow engine:
- business rule tasks
- user tasks
- script tasks
- send tasks
- receive task
- undefined tasks (manual tasks)

lenaschoenburg committed:
ci: allow using partial reports of failed version compatibility check runs

Even when a previous run fails (for example because it timed out),
we can use the report to continue where we left of.

Commit:
Merge branch 'be-6653-index-null-3' of github.com:camunda/zeebe into be-6653-index-null-3

lenaschoenburg committed:
refactor: remove remaining scala code (#17370)

This rewrites all remaining scala files in Java to simplify our build
process.

lenaschoenburg committed:
ci: allow using partial reports of failed version compatibility check runs (#17602)

## Description
Even when a previous run fails (for example because it timed out), we
can use the report to continue where we left of.
This will help the workflow to make progress now that we released new
versions:
https://github.com/camunda/zeebe/actions/workflows/zeebe-version-compatibility.yml

ce-dmelnych committed:
feat: extend execution listener support to all task types (#17372)

## Description
This Pull Request extends Execution Listener support across all task
types within the Zeebe workflow engine:
- business rule tasks
- user tasks
- script tasks
- receive tasks
- send tasks
- undefined tasks (manual tasks).

### Key Features
- Implement EL support for a wide range of task elements, enabling pre
and post execution custom logic.
- Ensure EL configuration support for each task type through the Java
builder API.

closes #16211

matt-whiteman committed:
test: refactor some Operate integration tests for efficiency (#17593)

## Description

Refactored ReindexIT and SchemaCreationIT into new templates

Removed several tests and test suites that seemed to not have much
value:

ElasticsearchConnectorBasicAuthIT - Many other tests connect to ES as
part of their setup already
ElasticsearchConnectorSSLAuthIT - Many other tests connect to ES as part
of their setup already
ZeebeConnectorIT - Many other tests connect to Zeebe as part of their
setup already, test takes very long to run
ZeebeConnectorSecureIT - Many other tests connect to Zeebe as part of
their setup already, test takes very long to run
IndexControllerIT - test has been disabled for almost 3 years
ShowErrorMessagesInDevelopmentIT - test has been disabled for a long
time and does not seem to have much value
SuppressErrorMessagesInProductionIT.java - test has been disabled for a
long time and does not seem to have much value

ReindexIT.logReindexProgress - Increased test time by over 60 seconds
and used dependency no longer updated just to test logging
ReindexIT.resetIndexSettings - test did not seem to do what it stated,
updating index settings is completely contained within elasticsearch and
has no operate logic

## Related issues

closes #

Commit:
Merge branch 'be-6653-index-null-3' of github.com:camunda/zeebe into be-6653-index-null-3

Commit:
Merge branch 'be-6653-index-null-3' of github.com:camunda/zeebe into be-6653-index-null-3
